metadata/rest/doi_queries/doi_registration_update.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""CherryPy DOI Registration Updater object class."""
+from __future__ import print_function
+from peewee import fn
+from cherrypy import tools, request, HTTPError
+import requests
+from dateutil.parser import parse
+from metadata.rest.user_queries.user_search import UserSearch
+from metadata.orm import DOIEntries, DOITransaction, TransactionRelease
+from metadata.orm import DOIAuthors, DOIAuthorMapping, DOIInfo
+from xml.etree import ElementTree
+import pprint
+
+
+class DOIRegistrationUpdate(object):
+    """Updates the database with new DOI registration info from registration API."""
+
+    exposed = True
+
+    @staticmethod
+    def _add_doi_record_entry(doi_info):
+        # translate existing user from network id
+        user_info = UserSearch.search_for_user(doi_info.get('owner_network_id'), 'simple').pop()
+        user_id = user_info.get('person_id')
+        meta_info = doi_info.get('meta')
+
+        infix_components = meta_info.get('doi_infix').split('.')
+        transaction_id = infix_components.pop()
+
+        # Check if transaction is released
+        tr_check_query = TransactionRelease().select().where(TransactionRelease.transaction == transaction_id)
+        if tr_check_query.count() == 0:
+            message = 'Transaction {0} has not been released'.format(transaction_id)
+            raise HTTPError('400 Bad Request', message)
+
+        # make sure this record doesn't already exist
+        # add doi_entry record
+        DOIRegistrationUpdate._update_doi_entry_info(
+            doi_info.get('doi'),
+            meta_info,
+            user_info.get('person_id')
+        )
+
+        # add doi to transaction mapping
+        doi_trans_map_item = {
+            'doi': doi_info.get('doi'),
+            'transaction': transaction_id
+        }
+        doi_trans_mapping, mapping_created = DOITransaction.get_or_create(**doi_trans_map_item)
+        return doi_trans_mapping.doi.doi
+
+    @staticmethod
+    def _get_updated_osti_info(doi_string, elink_url, elink_user, elink_password):
+        """This will access the server at OSTI and get all the relevant details for this DOI."""
+        osti_elink_url = elink_url + '?doi=' + doi_string
+
+        r = requests.get(osti_elink_url, auth=(elink_user, elink_password))
```

Comment:
I'm not sure I like this option, doesn't the DOI service already have this info from osti? Can't the DOI service send me this info from osti in the post?